reana_workflow_controller/rest.py

```diff
@@ -316,4 +339,171 @@ def run_yadage_workflow_from_spec_endpoint():  # noqa
+              workflow_id:
+                type: string
+          examples:
+            application/json:
+              {
+                "message": "Workflow successfully launched",
+                "workflow_id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed
+    """
+    try:
+        if request.json:
+            queue = organization_to_queue[request.args.get('organization')]
+            resultobject = run_cwl_workflow.apply_async(
+                args=[request.json],
+                queue='cwl-{}'.format(queue)
+            )
+            return jsonify({'message': 'Workflow successfully launched',
+                            'workflow_id': resultobject.id}), 200
+
+    except (KeyError, ValueError):
+        traceback.print_exc()
+        abort(400)
+
+
+@restapi_blueprint.route('/cwl/spec', methods=['POST'])
+def run_cwl_workflow_from_spec_endpoint():  # noqa
+    r"""Create a new cwl workflow.
+
+    ---
+    post:
+      summary: Creates a new cwl workflow from a specification file.
+      description: This resource is expecting a JSON cwl specification.
+      operationId: run_cwl_workflow_from_spec
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: organization
+          in: query
+          description: Required. Organization which the worklow belongs to.
+          required: true
+          type: string
+        - name: user
+          in: query
+          description: Required. UUID of workflow owner.
+          required: true
+          type: string
+        - name: workflow
+          in: body
+          description: >-
+            JSON object including workflow parameters and workflow
+            specification in JSON format (`cwlschemas.load()` output)
+            with necessary data to instantiate a cwl workflow.
+          required: true
+          schema:
+            type: object
+            properties:
+              parameters:
+                type: object
+                description: Workflow parameters.
+              workflow_spec:
+                type: object
+                description: >-
+                  cwl specification in JSON format.
+      responses:
+        200:
+          description: >-
+            Request succeeded. The workflow has been instantiated.
+          schema:
+            type: object
+            properties:
+              message:
+                type: string
+              workflow_id:
+                type: string
+          examples:
+            application/json:
+              {
+                "message": "Workflow successfully launched",
+                "workflow_id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed
+    """
```

Comment:
remember to amend this `docstring` accordingly so when you run `./run_tests.sh` you will obtain the updated version of your API using OpenApi spec and you can copy it to `reana-server`